tools/run_tests/python_utils/bazel_report_helper.py

```diff
@@ -0,0 +1,252 @@
+            'else',
+            '  # success: plant the pre-generated xml report that says "success"',
+            '  mv -f %s %s' %
+            (success_report_filename, failing_report_filename),
+            'fi',
+        ]
+
+        lines = [
+            line + '\n'
+            for line in intro_lines + upload_results_lines + outro_lines
+        ]
+        f.writelines(lines)
+    os.chmod(bazel_wrapper_filename, 0o775)  # make the unix wrapper executable
+
+    # generate bazel wrapper for windows
+    with open(bazel_wrapper_bat_filename, 'w') as f:
+        intro_lines = [
+            '@echo on',
+            '',
+            'bazel --bazelrc="%s" %%*' % bazel_rc_filename,
+            'set BAZEL_EXITCODE=%errorlevel%',
+            '',
+        ]
+
+        if upload_results:
+            upload_results_lines = [
+                '@rem Sleep to let ResultStore finish writing results before querying',
+                'sleep 60',
+                'python3 tools/run_tests/python_utils/upload_rbe_results.py --invocation_id="%s" || exit /b 1'
+                % invocation_id,
+                '',
+            ]
+        else:
+            upload_results_lines = []
+
+        outro_lines = [
+            'if %BAZEL_EXITCODE% == 0 (',
+            '  @rem success: plant the pre-generated xml report that says "success"',
+            '  mv -f %s %s' %
+            (success_report_filename, failing_report_filename),
+            ')',
+            'exit /b %BAZEL_EXITCODE%',
+        ]
+
+        lines = [
+            line + '\n'
+            for line in intro_lines + upload_results_lines + outro_lines
+        ]
+        f.writelines(lines)
+
+    print('Bazel invocation ID: %s' % invocation_id, file=sys.stderr)
+    print('Upload test results to BigQuery after bazel runs: %s' %
+          upload_results,
+          file=sys.stderr)
+    print('Generated bazel wrapper: %s' % bazel_wrapper_filename,
+          file=sys.stderr)
+    print('Generated bazel wrapper: %s' % bazel_wrapper_bat_filename,
+          file=sys.stderr)
+
+
+# parse command line
+argp = argparse.ArgumentParser(
+    description='Generate bazel wrapper to help with bazel test reports in CI.')
+argp.add_argument(
+    '--report_path',
+    required=True,
+    type=str,
+    help=
+    'Path under which the bazel wrapper and other files are going to be generated'
+)
+argp.add_argument('--report_suite_name',
+                  default='bazel_invocations',
+                  type=str,
+                  help='Test suite name to use in generated XML report')
+args = argp.parse_args()
+
+# generate new bazel invocation ID
+invocation_id = str(uuid.uuid4())
+
+report_path = args.report_path
+report_suite_name = args.report_suite_name
+upload_results = True if os.getenv('UPLOAD_TEST_RESULTS') else False
+
+_append_to_kokoro_bazel_invocations(invocation_id)
+_create_bazel_wrapper(report_path, report_suite_name, invocation_id,
+                      upload_results)
```

Comment:
Let's create a main function, and protect this file with:

```py
if __name__ == '__main__':
    # Execute when the module is not initialized from an import statement.
    ...
```

https://docs.python.org/3/library/__main__.html#idiomatic-usage

Reply:
Yup, good point, I'll do that.

Reply:
done.